### PR TITLE
Analyzer IP API

### LINKED ISSUE #1260: [Analyzer] IP API
## Name
ipapi

## Link
https://ip-api.com/

## Type of analyzer
IP-based. 2 possible endpoints:
* https://ip-api.com/docs/api:batch
* https://ip-api.com/docs/dns

## Why should we use it
free endpoint with plain HTTP and max 45 requests/minute

### PR DESCRIPTION
**Pull request by shivam-Purohit**
Added analyzer IP API closes #1260 

# Description

IP API gives intel about multiple IPs and also provides a DNS report 

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] A new plugin (analyzer, connector or playbook) was added or changed, in which case:
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [x] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] If a File analyzer was added, its name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
    - [x] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook in `playbook_config.json`
    - [x] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] The tests gave 0 errors.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](./Contribute.md)).
- [ ] If the GUI has been modified, I have a provided a screenshot of the result in the PR.

sample IP : 208.80.152.201 (no query passed)

JSON response : 
```
{
  'response_IP': [
    {
      'status': 'success',
      'country': 'United States',
      'countryCode': 'US',
      'region': 'IL',
      'regionName': 'Illinois',
      'city': 'Chicago',
      'zip': '60666',
      'lat': 41.8781,
      'lon': -87.6298,
      'timezone': 'America/Chicago',
      'isp': 'Wikimedia Foundation Inc.',
      'org': 'Wikimedia Foundation Inc',
      'as': 'AS14907 Wikimedia Foundation Inc.',
      'query': '208.80.152.201'
    }
  ],
  'resonse_DNS': {
    'dns': {
      'geo': 'India - Bharti Airtel',
      'ip': '122.185.151.250'
    }
  }
}
```
ubuntu  Running  - Oracle VM VirtualBox 06-02-2023 01_18_53

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### CODE REVIEW

**Comment by shivam-Purohit** [comment]
The query can handle upto 100 IPs but I am sure how to add support for more than one. A single value can be taken from the Observable_name. We can take multiple but that could just create new jobs.

File: configuration/analyzer_config.json [hunk 1]
```diff
@@ -1749,6 +1749,32 @@
       }
     }
   },
+  "IPApi": {
+    "type": "observable",
+    "python_module": "ipapi.IPApi",
+    "description": "Gives information about IPs and DNS",
```
Comment (mlodic) [inline configuration/analyzer_config.json:1755]: please add link to external service with markdown

**Review by mlodic — COMMENTED** [review]
seems good!

**Comment by shivam-Purohit** [comment]
there is some bug still persistent in the plugin section. Was trying to delete the parameter, but the app was not responding. This message pops up during the process.
ubuntu  Running  - Oracle VM VirtualBox 06-02-2023 19_23_05

File: configuration/analyzer_config.json (hunk 1 above)
Comment (shivam-Purohit) [reply configuration/analyzer_config.json]: @mlodic can you review it, please?

**Comment by mlodic** [comment]
> there is some bug still persistent in the plugin section. Was trying to delete the parameter, but the app was not responding. This message pops up during the process. ubuntu Running - Oracle VM VirtualBox 06-02-2023 19_23_05

without the error log, it is impossible to debug this. For these cases (found bugs) you need to get used to check application logs. There you can find the reason of that error. Please, try to do do that (it should be in `var/lib/docker/volumes/intelowl_generic_logs/_data/django` or something similar) and let's analyze that error together

**Comment by shivam-Purohit** [comment]
> 

I am trying that. Should I open a new issue regarding the bug or try in this PR only?

**Comment by mlodic** [comment]
that is something related to database caching which is difficult to understand from these logs only. Could you please open an issue regarding this?

**Comment by shivam-Purohit** [comment]
I was about to, but you opened it first. I will try to investigate the issue and come up with something.

**Pull request merged**